File: matlab/lib/+Ice/TwowayOnlyException.m
classdef (Sealed) TwowayOnlyException < Ice.LocalException
    %TWOWAYONLYEXCEPTION The exception that is thrown when attempting to invoke an operation with ice_oneway,
    %   ice_batchOneway, ice_datagram, or ice_batchDatagram, and the operation has a return value, an out parameter,
    %   or an exception specification.

    % Copyright (c) ZeroC, Inc.

    methods (Hidden)
        function obj = TwowayOnlyException(operation)
            if nargin == 0
                superArgs = {};
            else
                assert(nargin == 1, 'Invalid number of arguments');
                superArgs = {'Ice:TwowayOnlyException', sprintf('operation ''%s'' can only be invoked with a twoway proxy',...
                    operation)};
            end
            obj@Ice.LocalException(superArgs{:});
        end
    end
end

File: java/src/com.zeroc.ice/src/main/java/com/zeroc/Ice/TwowayOnlyException.java
// Copyright (c) ZeroC, Inc.

package com.zeroc.Ice;

/**
 * The exception that is thrown when attempting to invoke an operation with {@code ice_oneway}, {@code ice_batchOneway},
 * {@code ice_datagram}, or {@code ice_batchDatagram}, and the operation has a return value, an out parameter, or an
 * exception specification.
 */
public final class TwowayOnlyException extends LocalException {
    /**
     * Constructs a TwowayOnlyException with the operation name.
     *
     * @param operation the name of the two-way only operation
     */
    public TwowayOnlyException(String operation) {
        super(
            "Cannot invoke operation '"
                + operation
                + "' with a oneway, batchOneway, datagram, of batchDatagram proxy.");
        this.operation = operation;
    }

    @Override
    public String ice_id() {
        return "::Ice::TwowayOnlyException";
    }

    /** The name of the two-way only operation. */
    public final String operation;

    private static final long serialVersionUID = -7036652448391478186L;
}
